packages/components/src/ui/menu/component.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { contextConnect, useContextSystem } from '@wp-g2/context';
+import { cx } from '@wp-g2/styles';
+// eslint-disable-next-line no-restricted-imports
+import { Menu as ReakitMenu } from 'reakit';
+
+/**
+ * Internal dependencies
+ */
+import { usePopoverResizeUpdater } from '../popover/utils';
+import { View } from '../view';
+import { MenuContext } from './context';
+import * as styles from './styles';
+
+/**
+ * `Menu` is an actionable component that displays a list of actions, links, or informative content.
+ *
+ * @example
+ * ```jsx
+ * <Menu>
+ *  <MenuItem>...</MenuItem>
+ *  <MenuItem>...</MenuItem>
+ *  <MenuItem>...</MenuItem>
+ * </Menu>
+ * ```
+ *
+ * @see https://reakit.io/docs/menu/#menu
+ *
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<import('./types').Props, 'div'>} props
+ * @param {import('react').Ref<any>} forwardedRef
+ */
+function Menu( props, forwardedRef ) {
+	const { as, children, className, menu, ...otherProps } = useContextSystem(
```

Comment:
I still don't understand when this `menu` prop will be populated. Will `useMenuState` be called somewhere? cc @ItsJonQ

Reply:
It is called in `Dropdown`: 

https://github.com/ItsJonQ/g2/blob/main/packages/components/src/Dropdown/Dropdown.js#L29

Then passed to `Menu` via `DropdownMenu`: 

https://github.com/ItsJonQ/g2/blob/main/packages/components/src/Dropdown/DropdownMenu.js#L25

Reply:
So we would use it like this or `DropdownMenu` will be part of `@wordpress/components`?

```jsx
import { DropdownMenu } from '@wp-g2/components';
import { MenuItem } from '@wordpress/components';

<DropdownMenu>
  <MenuItem />
  <MenuItem />
</DropdownMenu>
```

Can we add stories for that?

Reply:
I was planning on adding stories for it when I opened the next PR in this series to add the `DropdownMenu` component. We can't add them yet because the context they use will be different due to the different packages.

Reply:
Got it! Does `DropdownMenu` depend on this PR? I feel like this PR can't be properly reviewed without `DropdownMenu`.

Reply:
Yeah, `DropdownMenu` does depend on this. I wanted to add them separately otherwise it was too big of a PR but I'm happy to add it in here if you think that's overall better.

Reply:
I see! I agree that it's better to have a separate PR for `DropdownMenu`. I'd suggest then removing everything that can only be used with `DropdownMenu` from the component in this PR (including the changes related to Reakit), and then cherry-pick it on the `DropdownMenu` PR. What do you think?

Reply:
Sure, I can remove the menu aware stuff from this PR.